portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,63 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.Marker;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Handles fetching and saving markers data. */
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /** Responds with a JSON array containing marker data. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<Marker> markers = new ArrayList<>();
+    Query query = new Query("Marker");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      double lat = (double) entity.getProperty("lat");
+      double lng = (double) entity.getProperty("lng");
+      String city = (String) entity.getProperty("city");
+
+      Marker marker = new Marker(lat, lng, city);
+      markers.add(marker);
```

Comment:
Nit: The extra variable doesn't add much to readability. You can do these in 1 line.